libyul/AsmParser.cpp

```diff
@@ -78,14 +108,74 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+langutil::Token Parser::advance()
+{
+	auto const token = ParserBase::advance();
+	if (m_useSourceLocationFrom == UseSourceLocationFrom::Comments)
+		fetchSourceLocationFromComment();
+	return token;
+}
+
+void Parser::fetchSourceLocationFromComment()
+{
+	solAssert(m_charStreamMap.has_value(), "");
+
+	if (m_scanner->currentCommentLiteral().empty())
+		return;
+
+	static std::regex const lineRE = std::regex(
+		R"~~~(.*\s*@src\s+(\d+):(-1|\d+):(-1|\d+)\s*)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+
+	auto const commentLocation = m_scanner->currentCommentLocation();
+	string_view text = m_scanner->currentCommentLiteral();
+	std::cmatch cm;
+	if (!std::regex_search(text.data(), text.data() + text.size(), cm, lineRE))
+		return;
+	solAssert(cm.size() == 4, "");
+
+	auto const sourceIndex = toUnsignedInt(cm[1].str());
+	auto const start = toInt(cm[2].str());
+	auto const end = toInt(cm[3].str());
+
+	if (!sourceIndex || !start || !end)
```

Comment:
Maybe better to construct the SourcLocation object and then call `isValid` on it?

Reply:
Ah sorry, these are optionals!